### ISSUE TITLE
[List builder] Single column variant 

### ISSUE THREAD

**Issue by davidicus** [carbon-design-system/carbon-addons-iot-react#1491]
> **Is this an extension to an existing component, pattern, or a brand new asset?**
> This is an EDS item.
> 
> **What is the rationale for this contribution.**
> This contribution allows users to populate a list of items from a second list of possible items, eg when selecting things from a catalog to proceed to checkout with, or sets of rules to apply to a given document.
> 
> The single column variant allows users to see only the 'selected' items, and uses a type ahead search to allow users to find and add things to the selection, valuable for tight spaces.
> 
> **Detail the intended behavior**
> 
> * A single list is rendered with a search bar just below the header, always exapanded.
>   ![image](https://media.github.ibm.com/user/684/files/a8afa080-200c-11ea-8c86-01c40396be94)
> * When a user begins to interact with this search bar, appropriate options will be presented beneath of items that match the given search criteria.
>   ![image](https://media.github.ibm.com/user/684/files/b06f4500-200c-11ea-9ff8-764065b21f53)
> * Each of those items will have a 20px black fill 'add' icon present (right aligned).
> * Selecting one of these items will add the item to the main list and remove it from the options in the search dropdown.
>   ![image](https://media.github.ibm.com/user/684/files/bbc27080-200c-11ea-870d-ce052a69763a)
> * Items in the list will have a right aligned 20px black fill 'minus' icon, which can be removed by clicking on them.
>   ![image](https://media.github.ibm.com/user/684/files/c11fbb00-200c-11ea-9864-fe07db7cfea0)
> 
> Found: https://ibm.invisionapp.com/share/U2NYW8D7DFV#/screens/319100293

**Comment by davidicus** [comment]
@kevinsperrine Can you verify that the list builder we implemented closes this issue out?

**Comment by kevinsperrine** [comment]
No, I did not implement the single-column variation--only the two column version, because 1) two-column was asked for in other issue and 2) we need two-column for the SelectUserModal in the RuleBuilder.
